File: src/models/Collaborator.ts
import Task from './Task';
import User from './User';

export class Collaborator extends User {
  isAdmin: boolean;
  tasks: Task[];

  constructor(name: string, birthday: Date, email: string, phone: string, tasks: Task[]) {
    super(name, birthday, email, phone);
    this.isAdmin = false;
    this.tasks = tasks;
  }
}

File: src/HigiaServer.Domain/Entities/Collaborator.cs
namespace HigiaServer.Domain.Entities;

public class Collaborator : BaseUserEntity
{
    protected Collaborator() { }

    public Collaborator(string name, string email, string phoneNumber, DateTime birthday, string passwordHash) :
        base(name, email, phoneNumber, birthday, passwordHash)
    {
        IsAdmin = false;
    }

    public List<Task>? Tasks { get; protected set; } = new();
}